lib/hls/hls_parser.js

```diff
@@ -997,6 +997,22 @@ shaka.hls.HlsParser = class {
         this.presentationTimeline_.setSegmentAvailabilityDuration(
             segmentAvailabilityDuration);
       }
+
+      if (!this.presentationTimeline_.isStartTimeLocked()) {
+        const streamInfos = Array.from(this.uriToStreamInfosMap_.values());
+        const activeStreamInfos =
+            streamInfos.filter((s) => s.stream.segmentIndex);
+        for (const streamInfo of activeStreamInfos) {
```

Comment:
It'd be more efficient if, instead of making a separate array twice above, you just iterated over the map values and checked for active-ness inside the loop.

```js
for (const streamInfo of this.uriToStreamInfosMap_.values()) {
  if (!streamInfo.stream.segmentIndex) {
    continue; // Not active.
  }
```

Reply:
Done!